src/workerd/api/r2-bucket.c++

```diff
@@ -559,6 +582,22 @@ jsg::Promise<kj::Maybe<jsg::Ref<R2Bucket::HeadResult>>> R2Bucket::put(jsg::Lock&
       KJ_IF_SOME(s, o.storageClass) {
         putBuilder.setStorageClass(s);
       }
+      KJ_IF_SOME(ssecKey, o.ssecKey) {
```

Comment:
Can you de-duplicate this code? Seems like an exact copy of the code used in initGetOptions?

Reply:
Split into separate function.